qemu/tests/virtio_mem_ignore_shared.py

```diff
@@ -0,0 +1,96 @@
+import ast
+import time
+
+from virttest import error_context
+from virttest import utils_test
+
+from virttest.utils_misc import normalize_data_size
+from virttest.virt_vm import VMMigrateTimeoutError
+
+from provider import virtio_mem_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Migrates a guest with virtio-mem device and x-ignore-shared enabled
+    1) Boot source and destination VMs with a virtio-mem device
+       and shared memory.
+    2) Enable x-ignore-shared capability in both hosts.
+    3) Check the capability is enabled.
+    4) Start stress tool in source VM.
+    5) Migrate VM.
+    6) Memory tool stress is stopped and migration completed.
+    7) Prove the total ram migrated is a little amount.
+    8) Resize the virtio-mem device in destination a couple of times.
+    9) Ascertain the virtio-mem device work properly.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    threshold = params.get_numeric("threshold", target_type=float)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    vm.wait_for_login()
+    # Stress in source VM
+    error_context.base_context("Install and compile stress tool",
+                               test.log.info)
+    test_mem = params.get_numeric("mem", target_type=float)
+    params["stress_args"] = '--cpu 4 --io 4 --vm 2 --vm-bytes %fM' % float(test_mem * 0.8)
+    try:
+        stress_test = utils_test.VMStress(vm, "stress", params)
+        error_context.context("Stress test start", test.log.info)
+        stress_test.load_stress_tool()
```

Comment:
We should clean up the environment however the case is passed or failed to keep the pure env for the next case

Reply:
Updated, thanks !